test/e2e/clusterctl_upgrade.go

```diff
@@ -343,13 +350,19 @@ func ClusterctlUpgradeSpec(ctx context.Context, inputGetter func() ClusterctlUpg
 		)
 		Expect(err).NotTo(HaveOccurred())
 
-		By("Upgrading providers to the latest version available")
+		By("Upgrading providers to the latest version available using contract or to custom providers version if specified")
 		clusterctl.UpgradeManagementClusterAndWait(ctx, clusterctl.UpgradeManagementClusterAndWaitInput{
```

Comment:
I think we have to make sure to pass in the right parameters here:
* If one of the providers is set only the providers
* If not => pass in the contract

Reply:
(I would currently expect the e2e test to fail as we pass in all fields)